dht.go

```diff
@@ -376,6 +376,10 @@ func makeDHT(h host.Host, cfg dhtcfg.Config) (*IpfsDHT, error) {
 // lookupCheck performs a lookup request to a remote peer.ID, verifying that it is able to
 // answer it correctly
 func (dht *IpfsDHT) lookupCheck(ctx context.Context, p peer.ID) error {
+	if dht.routingTable.Size() < dht.bucketSize {
+		// no need to be picky if we don't have enough peers
+		return nil
+	}
```

Comment:
Shouldn't this be after the peer has replied to the GetClosestPeers request? If the reply had 0 elements then we'd still add it because that's the best we've got, but we should wait not add the peer if the `GetClosestPeer` call fails.

Reply:
Yes, agreed! Moved it after we get a response and updated the tests.